certmgr/mgr/file.go

```diff
@@ -254,7 +254,7 @@ func ReadSpecFile(path string, defaults *ParsableSpecOptions) (*cert.Spec, error
 	}
 
 	s, err := cert.NewSpec(path, &spec.ParsableSpecOptions.SpecOptions, &spec.Authority.Authority, spec.Request, pkiStorage)
-	if err != nil {
+	if s != nil {
```

Comment:
Can you check if `err != nil` first?

Reply:
Hmm, since we're returning the error, maybe instead we should just check `if err == nil {` here and append the callback, then just continue  to return the error. Otherwise, not sure what we do in the `if err != nil {}` other than, I guess, wrap the error.

Reply:
I re-looked at this with a fresh set of eyes and while I think the original PR is cleaner, I can see where it does break commonly used patterns and thus might lead to confusion.

So what we can do is check if err is not nil and just return nil, err there, and then unconditionally add the callback and return s,nil at the end, as reaching the final return shows err is not nil, and makes it clear that we are returning without error.  Updating PR with this now.